client/web/src/enterprise/batches/batch-spec/edit/library/LibraryPane.tsx

```diff
@@ -104,15 +106,32 @@ export const LibraryPane: React.FunctionComponent<React.PropsWithChildren<Librar
         [animateContainer, animateContent, animateHeader, setDefaultCollapsed]
     )
 
+    const { search: searchQuery } = useLocation()
+    const updateTemplateWithQueryAndName = useCallback((template: string): string => {
+        if (searchQuery !== '') {
+            const parameters = new URLSearchParams(location.search)
+
+            const query = parameters.get('q')
+            const patternType = parameters.get('patternType')
+
+            if (query) {
+                const searchQuery = `${query} ${patternType ? `patternType:${patternType}` : ''}`
```

Comment:
this would add a space at the end if patternType is not set, might want to add the space to the patterntype templatestr instead 
```suggestion
                const searchQuery = `${query}${patternType ? ` patternType:${patternType}` : ''}`
```